src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -44,30 +44,53 @@ class ResourceService(private val resourceTypes: Map[ResourceTypeName, ResourceT
-    val ownerExists = policies.exists { case (_, membership) => membership.roles.contains(resourceType.ownerRoleName) && membership.memberEmails.nonEmpty }
-
-    if (!ownerExists) {
-      throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"Cannot create resource without at least 1 policy with ${resourceType.ownerRoleName.value} role and non-empty membership"))
+  /**
+    * Validates the resource first and if any validations fail, an exception is thrown with an error report that
+    * describes what failed.  If validations pass, then the Resource should be persisted.
+    * @param resourceType
+    * @param resourceId
+    * @param policiesMap
+    * @param userInfo
+    * @return Future[Resource]
+    */
+  def createResource(resourceType: ResourceType, resourceId: ResourceId, policiesMap: Map[AccessPolicyName, AccessPolicyMembership], userInfo: UserInfo): Future[Resource] = {
+    makeCreatablePolicies(policiesMap).flatMap { policies =>
+      validateCreateResource(resourceType, resourceId, policies, userInfo) match {
+        case Some(errorReport) => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "Cannot create resource", errorReport))
+        case None => persistResource(resourceType, resourceId, policies)
+      }
     }
+  }
 
+  /**
+    * This method only persists the resource and then overwrites/creates the policies for that resource.
+    * Be very careful if calling this method directly because it will not validate the resource or its policies.
+    * If you want to create a Resource, use createResource() which will also perform critical validations
+    * @param resourceType
+    * @param resourceId
+    * @param policies
+    * @return Future[Resource]
+    */
+  private def persistResource(resourceType: ResourceType, resourceId: ResourceId, policies: Set[PotentialAccessPolicy]) = {
     for {
-      // validate policies first, overwritePolicy below will do this again but we do it first to avoid rollback on errors
-      _ <- Future.traverse(policies) { case (_, policyMembership) =>
-        mapEmailsToSubjects(policyMembership.memberEmails).map { members: Map[WorkbenchEmail, Option[WorkbenchSubject]] =>
-          validatePolicy(resourceType, policyMembership, members)
-        }
-      }
-
       resource <- accessPolicyDAO.createResource(Resource(resourceType.name, resourceId))
+      _ <- Future.traverse(policies)(createOrUpdatePolicy(resource, _))
+    } yield resource
+  }
 
-      _ <- Future.traverse(policies) { case (accessPolicyName, accessPolicyMembership) =>
-        overwritePolicy(resourceType, accessPolicyName, resource, accessPolicyMembership)
-      }
-    } yield {
-      resource
-    }
+  private def validateCreateResource(resourceType: ResourceType, resourceId: ResourceId, policies: Set[PotentialAccessPolicy], userInfo: UserInfo): Option[Seq[ErrorReport]] = {
```

Comment:
could this return just a Seq instead of Option[Seq]? Note that `case Seq()` matches an empty seq.